5.5/run-mysqld.sh

```diff
@@ -34,8 +34,16 @@ EOSQL
 
 # The MYSQL_ROOT_PASSWORD is optional
 if [ -v MYSQL_ROOT_PASSWORD ]; then
-mysql $mysql_flags <<EOSQL
-    SET PASSWORD FOR 'root'@'%' = PASSWORD('${MYSQL_ROOT_PASSWORD}');
+  mysql $mysql_flags <<EOSQL
+    GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' IDENTIFIED BY '${MYSQL_ROOT_PASSWORD}';
+EOSQL
+else
+  # We do GRANT and DROP USER to emulate a DROP USER IF EXISTS statement
+  # http://bugs.mysql.com/bug.php?id=19166
+  mysql $mysql_flags <<EOSQL
+    GRANT USAGE ON *.* TO 'root'@'%';
+    DROP USER 'root'@'%';
```

Comment:
Yes, I took special care to figure out how to do that. Users are per host, so that 'root'@'%' != 'root'@'localhost' -- and removing one leaves the other untouched. AFAICT it exactly reverts the changes from https://github.com/openshift/mysql/pull/90#discussion-diff-38482279R38, disabling remote access while keeping localhost access without password.